gostream/ffmpeg/avcodec.go

```diff
@@ -0,0 +1,679 @@
+//go:build cgo && !android
+
+package ffmpeg
+
+//#include <libavformat/avformat.h>
+//#include <libavcodec/avcodec.h>
+//#include <libavcodec/packet.h>
+//#include <libavutil/avutil.h>
+//#include <libavutil/imgutils.h>
+//#include <libswscale/swscale.h>
+// static const char *error2string(int code) { return av_err2str(code); }
+import "C"
+
+import (
+	"context"
+	"image"
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// YUV420P the pixel format AV_PIX_FMT_YUV420P
+	YUV420P = C.AV_PIX_FMT_YUV420P
+	// Target bitrate in bits per second.
+	bitrate = 1_200_000
+	// Target bitrate tolerance factor.
+	// E.g., 2.0 gives 100% deviation from the target bitrate.
+	bitrateDeviation = 2
+)
+
+// From https://ffmpeg.org/doxygen/trunk/pixfmt_8h_source.html#l00064
+const (
```

Comment:
I still haven't found a good way to port over C enums, so I've resorted to doing this. Vim macros made this a lot faster than it looks.